sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableClientTest.java

```diff
@@ -1338,4 +1334,13 @@ public void allowDeleteEntityWithEmptyPrimaryKey() {
         tableClient.createEntity(entity);
         tableClient.deleteEntity(entity);
     }
+
+    @Test
+    void NonRuntimeExceptionsDontGetCasted() {
```

Comment:
Might want to add this into a test class that doesn't extend `TestProxyTestBase`